samples/webrequest/webrequest.cpp

```diff
@@ -0,0 +1,157 @@
+            new wxStaticBoxSizer(wxVERTICAL, getPanel, "Image");
+        m_getStaticBitmap = new wxStaticBitmap(getImageBox->GetStaticBox(),
+            wxID_ANY, wxArtProvider::GetBitmap(wxART_MISSING_IMAGE));
+        getImageBox->Add(m_getStaticBitmap, wxSizerFlags(1).Expand());
+        getSizer->Add(getImageBox, wxSizerFlags(1).Expand().Border());
+
+        getPanel->SetSizer(getSizer);
+        notebook->AddPage(getPanel, "GET Image", true);
+
+        // POST Text page
+        wxPanel* postPanel = new wxPanel(notebook);
+        wxSizer* postSizer = new wxBoxSizer(wxVERTICAL);
+        postSizer->Add(new wxStaticText(postPanel, wxID_ANY, "Request URL:"),
+            wxSizerFlags().Border());
+        m_postURLTextCtrl = new wxTextCtrl(postPanel, wxID_ANY,
+            "https://api.github.com/");
+        postSizer->Add(m_postURLTextCtrl,
+            wxSizerFlags().Expand().Border(wxLEFT | wxRIGHT));
+
+        postSizer->Add(new wxStaticText(postPanel, wxID_ANY, "Content type:"),
+            wxSizerFlags().Border());
+        m_postContentTypeTextCtrl = new wxTextCtrl(postPanel, wxID_ANY,
+            "application/json");
+        postSizer->Add(m_postContentTypeTextCtrl,
+            wxSizerFlags().Expand().Border(wxLEFT | wxRIGHT));
+
+        postSizer->Add(new wxStaticText(postPanel, wxID_ANY, "Request body:"),
+            wxSizerFlags().Border());
+        m_postRequestTextCtrl = new wxTextCtrl(postPanel, wxID_ANY, "",
+            wxDefaultPosition, wxDefaultSize, wxTE_MULTILINE);
+        postSizer->Add(m_postRequestTextCtrl,
+            wxSizerFlags(1).Expand().Border(wxLEFT | wxRIGHT));
+
+        wxButton* postSendButton = new wxButton(postPanel, wxID_ANY, "&Send");
+        postSendButton->Bind(wxEVT_BUTTON, &WebRequestFrame::OnPostSendButton, this);
+        postSizer->Add(postSendButton, wxSizerFlags().Border());
+
+        postSizer->Add(new wxStaticText(postPanel, wxID_ANY, "Response body:"),
+            wxSizerFlags().Border());
+        m_postResponseTextCtrl = new wxTextCtrl(postPanel, wxID_ANY, "",
+            wxDefaultPosition, wxDefaultSize, wxTE_MULTILINE | wxTE_READONLY);
+        m_postResponseTextCtrl->SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_BTNFACE));
+        postSizer->Add(m_postResponseTextCtrl,
+            wxSizerFlags(1).Expand().Border(wxLEFT | wxRIGHT | wxBOTTOM));
+
+        postPanel->SetSizer(postSizer);
+        notebook->AddPage(postPanel, "POST Text");
+
+        mainSizer->Add(notebook, wxSizerFlags(1).Expand().Border());
+
+        SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_BTNFACE));
+        SetSizer(mainSizer);
+
+        SetSize(FromDIP(wxSize(400, 500)));
+
+        CreateStatusBar();
+    }
+
+    void OnGetLoadButton(wxCommandEvent& WXUNUSED(evt))
+    {
+        GetStatusBar()->SetStatusText("Requesting image...");
+
+        wxWebRequest* request = wxWebSession::GetDefault().CreateRequest(m_getURLTextCtrl->GetValue());
```

Comment:
Who is going to do the memory management? Should the user delete it / use a shared pointer or will wxWidgets take ownership?

Reply:
When I'm further with the implementation that will be cleaned up a little, in code and in documentation.